src/django/api/facility_history.py

```diff
@@ -0,0 +1,126 @@
+                entry.prev_record.location.y,
+            ],
+        },
+        'new': {
+            'type': 'Point',
+            'coordinates': [
+                entry.location.x,
+                entry.location.y,
+            ],
+        },
+    }
+
+
+def create_facility_history_dictionary(entry):
+    updated_at = str(entry.history_date)
+    change_reason = entry.history_change_reason or ''
+    history_type_display = entry.get_history_type_display()
+
+    if 'FacilityLocation' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'UPDATE',
+            'changes': {
+                'location': create_location_diff_from_history_entry(entry),
+            },
+        }
+    elif 'Merged with' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'MERGE',
+            'detail': change_reason,
+        }
+    elif 'Promoted' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'UPDATE',
+            'changes': {
+                'location': create_location_diff_from_history_entry(entry),
+                'name': {
+                    'old': entry.prev_record.name,
+                    'new': entry.name,
+                },
+                'address': {
+                    'old': entry.prev_record.address,
+                    'new': entry.address,
+                },
+            },
+        }
+    elif 'Created' in history_type_display:
+        return {
+            'updated_at': updated_at,
+            'action': 'CREATE',
+        }
+    elif 'Deleted' in history_type_display:
+        return {
+            'updated_at': updated_at,
+            'action': 'DELETE',
+        }
+    else:
+        return {
+            'updated_at': updated_at,
+            'action': history_type_display,
+        }
+
+
+def maybe_get_split_action_time_from_processing_results(item):
+    split_processing_times = [
+        r.get('finished_at', None)
+        for r
+        in item.processing_results
+        if r.get('action', None) == ProcessingAction.SPLIT_FACILITY
+    ]
+
+    return next(iter(split_processing_times), None)
+
+
+def processing_results_has_split_action_for_oar_id(list_item, facility_id):
+    return facility_id in [
+        r.get('previous_facility_oar_id', None)
+        for r
+        in list_item.processing_results
+        if r.get('action', None) == ProcessingAction.SPLIT_FACILITY
+    ]
+
+
+def create_facility_history_list(entries, facility_id):
+    facility_split_entries = [
+        {
+            'updated_at': maybe_get_split_action_time_from_processing_results(
+                m.facility_list_item
+            ),
+            'action': 'SPLIT',
+            'detail': '{} was split from {}'.format(
+                m.facility.id,
+                facility_id,
+            )
+        }
+        for m
+        in FacilityMatch
+        .objects
+        .filter(confidence=1.0)
```

Comment:
HM. I added the `confidence=1.0` and `status=FacilityMatch.CONFIRMED` filters here so the query would narrow down the list of matches it looks through before doing the if statement. Since I think prod would have tens or hundreds of thousands of match rows to run through.

We directly set both of these in the `split` endpoint implementation. We could make `1.0` into a constant that gets used in both places?

Reply:
Thanks for clarifying the reason for the filters.

I did a little reading about JSONB searching and found that this query which should get us what we want.

```sql
SELECT * 
FROM api_facilitylistitem 
WHERE processing_results @> '[{"action": "split_facility"}]';
```

What do you think about this?

```diff
diff --git a/src/django/api/facility_history.py b/src/django/api/facility_history.py
index d17d5c3..8ccc511 100644
--- a/src/django/api/facility_history.py
+++ b/src/django/api/facility_history.py
@@ -1,3 +1,5 @@
+from django.db.models import F
+
 from api.constants import ProcessingAction
 from api.models import FacilityMatch
 
@@ -108,8 +110,10 @@ def create_facility_history_list(entries, facility_id):
         for m
         in FacilityMatch
         .objects
-        .filter(confidence=1.0)
-        .filter(status=FacilityMatch.CONFIRMED)
+        .annotate(
+            processing_results=F('facility_list_item__processing_results'))
+        .extra(
+            where=['processing_results @> \'[{"action": "split_facility"}]\''])
         if processing_results_has_split_action_for_oar_id(
             m.facility_list_item,
             facility_id,
```

Reply:
Cool -- that's better. I made this change in f7f97315aed5b0e69f0852d30acd9fa566e3c867